streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractKTableKTableJoinValueGetterSupplier.java

```diff
@@ -32,14 +30,10 @@ public AbstractKTableKTableJoinValueGetterSupplier(final KTableValueGetterSuppli
     public String[] storeNames() {
         final String[] storeNames1 = valueGetterSupplier1.storeNames();
         final String[] storeNames2 = valueGetterSupplier2.storeNames();
-        final ArrayList<String> stores = new ArrayList<>(storeNames1.length + storeNames2.length);
-        for (final String storeName : storeNames1) {
-            stores.add(storeName);
-        }
-        for (final String storeName : storeNames2) {
-            stores.add(storeName);
-        }
-        return stores.toArray(new String[stores.size()]);
+        final String[] stores = new String[storeNames1.length + storeNames2.length];
+        System.arraycopy(storeNames1, 0, stores, 0, storeNames1.length);
+        System.arraycopy(storeNames2, 0, stores, storeNames1.length, storeNames2.length);
+        return stores;
```

Comment:
@ijuma @original-brownbear Yes. It's an array because it's passed into `TopologyBuilder#connectProcessorAndStateStores(String processorName, String... stateStoreNames)`